x/feegrant/filtered_fee_test.go

```diff
@@ -0,0 +1,208 @@
+			accept:    false,
+		},
+		"small fee without expire": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+			},
+			msgs:    []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			fee:     smallAtom,
+			accept:  true,
+			remove:  false,
+			remains: leftAtom,
+		},
+		"all fee without expire": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: smallAtom,
+			},
+			msgs:   []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			fee:    smallAtom,
+			accept: true,
+			remove: true,
+		},
+		"wrong fee": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: smallAtom,
+			},
+			msgs:   []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			fee:    eth,
+			accept: false,
+		},
+		"non-expired": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+				Expiration: &oneHour,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       smallAtom,
+			blockTime: now,
+			accept:    true,
+			remove:    false,
+			remains:   leftAtom,
+		},
+		"expired": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+				Expiration: &now,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       smallAtom,
+			blockTime: oneHour,
+			accept:    false,
+			remove:    true,
+		},
+		"fee more than allowed": {
+			allowance: &feegrant.BasicAllowance{
+				SpendLimit: atom,
+				Expiration: &oneHour,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       bigAtom,
+			blockTime: now,
+			accept:    false,
+		},
+		"with out spend limit": {
+			allowance: &feegrant.BasicAllowance{
+				Expiration: &oneHour,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       bigAtom,
+			blockTime: now,
+			accept:    true,
+		},
+		"expired no spend limit": {
+			allowance: &feegrant.BasicAllowance{
+				Expiration: &now,
+			},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			valid:     true,
+			fee:       bigAtom,
+			blockTime: oneHour,
+			accept:    false,
+		},
+	}
+
+	for name, stc := range cases {
+		tc := stc // to make scopelint happy
+		t.Run(name, func(t *testing.T) {
+			err := tc.allowance.ValidateBasic()
+			require.NoError(t, err)
+
+			ctx := app.BaseApp.NewContext(false, ocproto.Header{}).WithBlockTime(tc.blockTime)
+
+			// create grant
+			createGrant := func() feegrant.Grant {
+				var granter, grantee sdk.AccAddress
+				allowance, err := feegrant.NewAllowedMsgAllowance(tc.allowance, tc.msgs)
+				require.NoError(t, err)
+				grant, err := feegrant.NewGrant(granter, grantee, allowance)
+				require.NoError(t, err)
+				return grant
+			}
+			grant := createGrant()
+
+			// create some msg
+			call := feegrant.MsgRevokeAllowance{
+				Granter: "",
+				Grantee: "",
+			}
```

Comment:
related to above: I would use Accept on a more conventional use case: MsgSend or MsgVote or MsgDelegate